### ISSUE TITLE
How is this built on Linux?

### ISSUE THREAD

**Issue by Melab** [dotnet/roslyn#19492]
The document for cross platform compilation mentions prebuilt tools, but doesn't give a link to them. So, how do I build this on Linux?

**Comment by jmarolf** [comment]
@Melab make should download them.  Are you following the directions [here](https://github.com/dotnet/roslyn/blob/master/docs/infrastructure/cross-platform.md)?

**Comment by Melab** [comment]
@jmarolf That page says:

    Patching the right Mono version and acquiring all the tools can be very difficult, so we've saved pre-built versions on Azure.

It doesn't mention these pre-built versions again.

**Comment by jmarolf** [comment]
> Running make should download all these toolset binaries and kick off a build using MSBuild running on Mono.

the `toolset binaries` being referred to here are the ones stored on azure.

**Comment by jcouv** [comment]
I built on Linux (Docker) just last week using the instructions from the ["building on Linux"](https://github.com/dotnet/roslyn/blob/master/docs/infrastructure/cross-platform.md) page.

Here are the notes I took:
1. Enlist to Roslyn repo on Linux (for this I used Docker and `docker run -it ubuntu bash`, then installed git with `apt-get update`, `apt-get install git`, then `git clone https://github.com/dotnet/roslyn.git`)
3. From a clean repo (`git clean -xdf`), select an SDK with configuration in `build/Targets/Tools.props` and `global.json`. We'll use "2.0.0" and "2.1.1-preview-007094".
4. Obtain dotnet by installing [pre-requisites](https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x) (`apt-get install libunwind8 liblttng-ust0 libcurl3 libssl1.0.0 libuuid1  zlib1g libicu55 curl` for Ubuntu 16) then `./build/scripts/obtain_dotnet.sh`, then update your PATH with `PATH=$PATH:./Binaries/dotnet-cli/`
5. Restore with `./build/scripts/restore.sh`
6. Build with `dotnet build CrossPlatform.sln`

@Melab Let me know if we can close this issue. Thanks
